docs/src/modules/components/MarkdownDocs.js

```diff
@@ -70,7 +71,13 @@ function MarkdownDocs(props) {
           }
 
           if (renderedMarkdownOrDemo.component) {
-            const Component = markdownComponents[renderedMarkdownOrDemo.component];
+            const name = renderedMarkdownOrDemo.component;
+            const Component = componentComponents?.[name];
+
+            if (Component === undefined) {
+              throw new Error('Missing component');
+            }
```

Comment:
Why this condition is needed?

Reply:
The demo have a detailed error 

https://github.com/mui/material-ui/blob/cceb149cdd3de798f1555f84d6fc12e011d6f34a/docs/src/modules/components/MarkdownDocs.js#L91

For me it can occur if you try using a component that does not exist at all in the `src` folder.
The goal of the issue being to help find the source of the error.
But the current message is probably very bad.

How about the following ?

```js
throw new Error(`No component found at the path ${path.join('docs/src', name)}`);
```

Reply:
Nice!